packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -36,30 +38,67 @@ class ChartConnection extends DisposableController
   );
 
   Timer? _pollingTimer;
-  bool _connected = false;
 
-  late final isDeviceAndroid =
-      serviceConnection.serviceManager.vm?.operatingSystem == 'android';
+  /// If completed, this instance was connected to the application.
+  final Completer<void> _connected = Completer();
+
+  /// If true, the instance lost connection to the application.
+  final Completer<void> _disconnected = Completer();
+
+  late final isDeviceAndroid = _isDevToolsCurrentlyConnected()
+      ? serviceConnection.serviceManager.vm?.operatingSystem == 'android'
+      : false;
+
+  /// True if DevTools is in connected mode and the connection to the app is alive.
+  bool _isDevToolsCurrentlyConnected() =>
+      // Theoretically it should be enough to check only connectedState.value.connected,
+      // but practically these values are not always in sync and, if at least
+      // on of them means disconnection this class consider the connection as lost,
+      // and stops interaction with it.
+      !offlineDataController.showingOfflineData.value &&
+      serviceConnection.serviceManager.connectedState.value.connected &&
+      serviceConnection.serviceManager.connectedApp != null;
 
   Future<void> maybeConnect() async {
```

Comment:
can we rename this to `initialize`? The "connect" term is overloaded with DevTools being connected to an app. This is really a method to initialize the chart listeners. So renaming this to `intiailze()` and having a single completer `_initialized` would make this code less confusing.